deepchem/models/torch_models/dtnn.py

```diff
@@ -0,0 +1,276 @@
+        if mode not in ['regression']:
+            raise ValueError("Only 'regression' mode is currently supported")
+        self.n_tasks = n_tasks
+        self.n_embedding = n_embedding
+        self.n_hidden = n_hidden
+        self.n_distance = n_distance
+        self.distance_min = distance_min
+        self.distance_max = distance_max
+        self.step_size = (distance_max - distance_min) / n_distance
+        self.steps = np.array(
+            [distance_min + i * self.step_size for i in range(n_distance)])
+        self.steps = np.expand_dims(self.steps, 0)
+        self.output_activation = output_activation
+        self.mode = mode
+        self.dropout = dropout
+
+        # get DTNNEmbedding
+        self.dtnn_embedding = layers.DTNNEmbedding(n_embedding=self.n_embedding)
+
+        # get DTNNSteps
+        self.dtnn_step_1 = layers.DTNNStep(n_embedding=self.n_embedding,
+                                           n_distance=self.n_distance)
+
+        self.dtnn_step_2 = layers.DTNNStep(n_embedding=self.n_embedding,
+                                           n_distance=self.n_distance)
+
+        # get DTNNGather
+        self.dtnn_gather = layers.DTNNGather(
+            n_embedding=self.n_embedding,
+            layer_sizes=[self.n_hidden],
+            n_outputs=self.n_tasks,
+            output_activation=self.output_activation)
+
+        # get Final Linear Layer
+        self.linear = nn.LazyLinear(self.n_tasks)
+
+    def forward(self, inputs: List[torch.Tensor]):
+        """
+        Parameters
+        ----------
+        inputs: List
+            List of Tensors containing. (In the specified order)
+            - atom_number
+            - distance
+            - atom_membership
+            - distance_membership_i
+            - distance_membership_j
+
+        Returns
+        -------
+        output: torch.Tensor
+            Predictions of the Molecular Energy.
+
+        """
+        dtnn_embedding = self.dtnn_embedding(inputs[0])
+
+        dtnn_embedding = nn.Dropout(self.dropout)(dtnn_embedding)
+        dtnn_step = self.dtnn_step_1(
+            [dtnn_embedding, inputs[1], inputs[3], inputs[4]])
+
+        dtnn_step = nn.Dropout(self.dropout)(dtnn_step)
+        dtnn_step = self.dtnn_step_2(
+            [dtnn_step, inputs[1], inputs[3], inputs[4]])
+
+        dtnn_step = nn.Dropout(self.dropout)(dtnn_step)
+        dtnn_gather = self.dtnn_gather([dtnn_step, inputs[2]])
+
+        dtnn_gather = nn.Dropout(self.dropout)(dtnn_gather)
+        output = self.linear(dtnn_gather)
+        return output
+
+
+class DTNNModel(TorchModel):
+    """Implements DTNN models for regression.
+
+    This class implements the Directed Message Passing Neural Network (D-MPNN) [1]_.
+
+    Examples
+    --------
+    >>> import deepchem as dc
+    >>> import os
+    >>> from scipy import io as scipy_io
+    >>> from deepchem.data import NumpyDataset
+    >>> model_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
```

Comment:
The usage example should use a featurizer and show how the user can invoke this model on a new inputs. We shouldn't use direct matrix loading

Reply:
Added those, You may look into it.